src/reducer/WishlistReducer.jsx

```diff
@@ -0,0 +1,20 @@
+import {addToWishlist,removeFromWishlist} from "utils/wishlist";
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case  "ADD_TO_WISHLIST":
+      return {
+        ...state,
+        wishlist: addToWishlist(state.wishlist, action.payload),
```

Comment:
i guess you should use ... before wishlist state to make sure it's being cloned while passing arguments.
Not sure though, you can confirm.

Reply:
not needed